src/NzbDrone.Core/MediaFiles/DownloadedEpisodesCommandService.cs

```diff
@@ -48,9 +54,49 @@ private List<ImportResult> ProcessDroneFactoryFolder()
             return _downloadedEpisodesImportService.ProcessRootFolder(new DirectoryInfo(downloadedEpisodesFolder));
         }
 
+        private List<ImportResult> ProcessFolder(DownloadedEpisodesScanCommand message)
+        {
+            if (!_diskProvider.FolderExists(message.Path))
+            {
+                _logger.Warn("Folder specified for import scan [{0}] doesn't exist.", message.Path);
+                return new List<ImportResult>();
+            }
+
+            if (message.DownloadClientId.IsNotNullOrWhiteSpace())
+            {
+
+                var trackedDownload = _trackedDownloadService.Find(message.DownloadClientId);
```

Comment:
The ID needed here is:
`downloadClient.Id + "-" + downloadItem.DownloadId,`

I ran into this issue fixing the failed download handling so it would actually delete from the download client.

TL/DR: its always going to be null